pkg/compactor/compactor.go

```diff
@@ -665,6 +674,9 @@ func (c *MultitenantCompactor) compactUsers(ctx context.Context) {
 				// We don't want to count shutdowns as failed compactions because we will pick up with the rest of the compaction after the restart.
 				level.Info(c.logger).Log("msg", "compaction for user was interrupted by a shutdown", "user", userID)
 				return
+			case errors.Is(err, syscall.ENOSPC):
```

Comment:
Have you done any manual test to ensure the error is correctly captured, given it's something we can't easily / reliably assert with a unit test?

Reply:
I added a comment documenting it: https://github.com/grafana/mimir/pull/8237/#issuecomment-2144824839